README.md

```diff
@@ -654,7 +716,7 @@ While some resources are centralized (e.g. access to a server), others are unbou
 
 Every resource type SHOULD have a canonical location where its revocations are kept. This list is non-exclusive, and revocation messages MAY be gossiped between peers in a network to share this information more quickly.
 
-It is RECOMMENDED that the canonical revocation store be kept as close to (or inside) the resource it is about as possible. For example, the WebNative File System maintains a Merkle tree of revoked CIDs at a well-known path. Another example is that a centralized server could have an endpoint that lists the revoked UCANs by CID.
+It is RECOMMENDED that the canonical revocation store be kept as close to (or inside) the resource it is about as possible. For example, the WebNative File System maintains a Merkle tree of revoked CIDs at a well-known path. Another example is that a centralized server could have an endpoint that lists the revoked UCANs by [canonical CID].
```

Comment:
This may be controversial, but I can't think of another way to make this work. 

To solve #112, we need a stable identifier. With something more general like ucan-ipld is that we have multiple posisble CIDs for the same data. This means that at minimum what gets signed over and CID need to match for the canonical CID. So that not every implementation has to support IPLD (or other formats e.g. RDF), we already require signing over a JWT (unless you don't care about consistency with other implementations, but then you're alread off-spec).

This means that the CID needs to be be pretty specific. CIDv1 + base32 + SHA256 seems like a pretty safe bet, but I'm absolutley open to other options.

@cdata do you have any opinions? In #112 you mention base64 — I'm happy to use that instead. Base32 means that you can pull up the CID in a URL via IPFS (for those using IPFS), but it's obvioulsy longer.

Reply:
I'm not sure I understand the problem. Even though UCAN can be encoded in various formats issuer does sign over a payload in a specific encoding. Multihash of that payload is likely the best key for the revocation as it would not matter what the encoding of delegation is it's signature payload hash will remain the same.

Now that said unfortunately current UCAN format makes this bit trick and annoying unlike invocation format where task CID is explicitly captured inside an invocation. I have been wondering if we should do something similar with UCANs as well which would make revocation lookups simpler.

Reply:
The problem: since a UCAN can be respresented by multiple types of CID (e.g. CIDv0, CIDv1, etc), I have to at least parse every CID and extract just the hash, though this is potentially a lot of work on receipt of any UCAN. Restricting to a specific CID format, you can "just" match strings, but it feels a bit heavy-handed. I remember Juan once saying that CIDs aren't an interoperability mechanism between systems (to choose hashes etc), but rather a way to do upgrades with backwards compatability, and that in most cases applications should just stick with one CID format.

Reply:
What I meant to say is if UCAN had a following structure:

```haskell
type struct UCAN {
  payload: Link<UCANPayload>
  sig: VarSig
}
```

It would not matter what format you got the actual UCAN in because issuer would have signed over whatever the link in whatever encoding they chose. It does mean that you're facing that you may now get arbitrary hashes and formats, but if you don't support those algorithms well you won't be able to verify those UCANs anyway so revocations are probably not a concern, if you do support them you're back to comparing stings. And because we embraced CIDs base encoding is no longer relevant. Recepient can transcode key into base encoding that matches their store encoding of keys.